test/python/WMComponent_t/JobCreator_t/JobCreator_t.py

```diff
@@ -300,6 +300,50 @@ def testVerySimpleTest(self):
 
         return
 
+    def testCampaignName(self):
+        """
+        Test campaign name is written into job pickle file
+        """
+        myThread = threading.currentThread()
+
+        config = self.getConfig()
+
+        name = makeUUID()
+        nSubs = 1
+        nFiles = 1
+        workloadName = 'TestWorkload'
+
+        dummyWorkload = self.createWorkload(workloadName=workloadName)
+        workloadPath = os.path.join(self.testDir, 'workloadTest', 'TestWorkload', 'WMSandbox', 'WMWorkload.pkl')
+
+        self.createJobCollection(name=name, nSubs=nSubs, nFiles=nFiles, workflowURL=workloadPath)
+
+        testJobCreator = JobCreatorPoller(config=config)
+
+        # First, can we run once without everything crashing?
+        testJobCreator.algorithm()
+
+        getJobsAction = self.daoFactory(classname="Jobs.GetAllJobs")
+        result = getJobsAction.execute(state='Created', jobType="Processing")
+
+        # Find the test directory
+        testDirectory = os.path.join(self.testDir, 'jobCacheDir', 'TestWorkload', 'ReReco')
+        groupDirectory = os.path.join(testDirectory, 'JobCollection_1_0')
+
+        # Get job pickle file
+        jobDir = os.listdir(groupDirectory)[0]
+        jobFile = os.path.join(groupDirectory, jobDir, 'job.pkl')
+        self.assertTrue(os.path.isfile(jobFile))
+        f = open(jobFile, 'rb')
```

Comment:
Please use a context manager to ensure that the file descriptor gets closed. I.e.: `with open(...)`